fmf/utils.py

```diff
@@ -280,17 +299,41 @@ def check_dimension(dimension, values):
         # Every value must match at least one value for data
         return all([check_value(dimension, value) for value in values])
 
+    def check_name(pattern: str) -> bool:
+        """
+        Check whether the node name matches
+
+        Search for regular expression pattern if `regexp` is turned on,
+        simply compare strings otherwise.
+        """
+        # Node name has to be provided if name search requested
+        if name is None:
+            raise FilterError(
+                f"Filter by name '{pattern}' requested "
+                f"but node name not provided to 'filter()'.")
+
+        if regexp:
+            return bool(re.search(pattern, name))
+        else:
+            return pattern == name
+
     def check_clause(clause):
         """ Split into literals and check whether all match """
         # E.g. clause = 'tag: A, B & tag: C & tag: -D'
         # Split into individual literals by dimension
         literals = dict()
         for literal in re.split(r"\s*&\s*", clause):
```

Comment:
I'm trying to understand how is the escaping of the `&` and `|` supposed to work in the end. Can't we just require the AND and OR operators to have whitespaces around them?
```suggestion
        for literal in [part.strip() for subpart in
               clause.split(' & ') for part in subpart.split(' | ')]:
```

Reply:
the splits in list comprehension seems to also take only half as much time as re.split btw.

Reply:
L361 would be 
```
    return any([check_clause(clause)
                for clause in filter.split(' | ')])
```

Reply:
> I'm trying to understand how is the escaping of the `&` and `|` supposed to work in the end. Can't we just require the AND and OR operators to have whitespaces around them?

The plan is to use the backslash character `\` to escape them. The white space is ignored, this is the documeneted behaviour and has been working like this for some time already, so it would not be a good idea to change this as it would break existing configs. Let's continue discussion in #220.